### PR TITLE
Remove asCommutativeRing from Heyting

### PR DESCRIPTION
**Pull request by johnynek**
Addresses the issue in #89

### CODE REVIEW

**Comment by non** [comment]
:+1:

File: core/src/main/scala/algebra/lattice/Bool.scala [hunk 1]
```diff
@@ -1,31 +1,45 @@
...
+   * Every Boolean algebra is a CommutativeRing, but we don't extend
+   * CommutativeRing because, e.g. we might want a Bool[Int] and CommutativeRing[Int] to
+   * refer to different structures, by default.
+   */
+  def asCommutativeRing: CommutativeRing[A] =
+    new CommutativeRing[A] {
+      def zero: A = self.zero
+      def one: A = self.one
+      def plus(x: A, y: A): A = self.xor(x, y)
+      def negate(x: A): A = self.complement(x)
```
Comment (TomasMikula) [inline core/src/main/scala/algebra/lattice/Bool.scala]: `negate` here is still broken. It should be

```scala
def negate(x: A): A = x
```

Comment (non) [inline core/src/main/scala/algebra/lattice/Bool.scala]: Aha, right. I'll fix that now. Thanks!

**Comment by TomasMikula** [comment]
Also note that if we had a type `BoolRing`, it could implement `negate` (to be `negate(x) = x`) and then the instance returned from `Bool#asCommutativeRing` would not have to override `negate`.

**Comment by non** [comment]
Let's open an issue about possibly adding `BoolRing`, and I will submit a PR to fix this bug. (I also want to ensure that we have a test that detects this issue.)

**Comment by TomasMikula** [comment]
Sounds good, I will open the issue.

**Pull request merged**